addon/HomepageStudy.jsm

```diff
@@ -0,0 +1,83 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+"use strict";
+
+const {classes:Cc, interfaces: Ci, utils: Cu} = Components;
+
+const EXPORTED_SYMBOLS = ["HomepageStudy"];
+
+Cu.import("resource://gre/modules/Services.jsm");
+Cu.import("resource://gre/modules/Console.jsm");
+
+// TODO: Change this for a proper path.
+const RAPPOR_PATH = `jar:file:///Users/arodriguez/src/shield/shield-study-rappor/dist/addon.xpi!/bootstrap.js/.././TelemetryRappor.jsm`;
+const { TelemetryRappor } = Cu.import(RAPPOR_PATH, {});
+
+const console = new ConsoleAPI({prefix: "shield-study-rappor"});
+
+const PREF_HOMEPAGE = "browser.startup.homepage";
+
+/**
+ * @returns the eTLD+1 of the user's homeapage. If the homepage is about:home,
+ * this value is returned. If it's any other about page, about:pages is returned.
+ * If the return value is null, there is a error with the homepage.
+ */
+function getHomepage(){
+
+  let homepage;
+  try {
+    homepage = Services.prefs.getComplexValue(PREF_HOMEPAGE, Ci.nsIPrefLocalizedString).data;
+  } catch (e) {
+    console.error("Error obtaining the homepage: ", e);
+    return null;
+  }
+  // transform the homepage into a nsIURI. Neccesary to get the base domain
+  let homepageURI;
+  try {
+    let uriFixup = Cc["@mozilla.org/docshell/urifixup;1"].getService(Ci.nsIURIFixup);
+    homepageURI = uriFixup.createFixupURI(homepage, Ci.nsIURIFixup.FIXUP_FLAG_NONE);
+  } catch (e) {
+    console.error("Error creating URI from homepage string: ", e);
+    return null;
+  }
+
+  let eTLD;
+  if (homepage.startsWith("about:")) {
```

Comment:
I think this could be a bit clearer if we just return early instead of setting the `eTLD` variable (see the code style's [general practices](https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style#General_C.2FC.2B.2B_Practices). We could change this to something like:

```
if (homepage.startsWith("about:")) {
  return (homepage == "about:home") ? homepage : "about:pages";
}

let eTLD;
try {
  // Do stuff here
} ...
```